data_model/src/events/mod.rs

```diff
@@ -65,6 +68,26 @@ pub mod model {
         Time(time::TimeEventFilter),
         /// Listen to trigger execution event with filter.
         ExecuteTrigger(execute_trigger::ExecuteTriggerEventFilter),
+        /// Listen to notifications event with filter.
+        Notification(notification::NotificationEventFilter),
+    }
+
+    /// Event filter which could be attached to trigger.
+    #[allow(variant_size_differences)]
+    #[derive(
+        Debug, Clone, PartialEq, Eq, FromVariant, Decode, Encode, Deserialize, Serialize, IntoSchema,
+    )]
+    // TODO: Temporarily made opaque
+    #[ffi_type(opaque)]
+    pub enum TriggerableFilterBox {
```

Comment:
Also suggest to rename to `TriggeringFilterBox`. So that it means that this filter can trigger, not to be triggered